### PR TITLE
[inductor] Fix MKL issue with test_indirect_device_assert

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108172

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @anijain2305

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -7543,6 +7544,7 @@ def fn(x: torch.Tensor) -> torch.Tensor:
                     fn_opt(inps)
 
         @skipIfRocm
+        @skip_if_consumer_card
```
Comment (peterbell10) [inline test/inductor/test_torchinductor.py:7547]: The test works for me on a 2060. What's issues are you seeing on a 3090?

Comment (jansel) [reply test/inductor/test_torchinductor.py:7547]: Subprocess is failing with:
```
b'Error: mkl-service + Intel(R) MKL: MKL_THREADING_LAYER=INTEL is incompatible with libgomp.so.1 library.\n\tTry to import numpy first or set the threading layer accordingly. Set MKL_SERVICE_FORCE_INTEL to force it.\n'
```

Actually might not be related to consumer cards, that was a guess based on test names.  I think I can fix it with `MKL_THREADING_LAYER=GNU`.

Comment (peterbell10) [reply test/inductor/test_torchinductor.py]: That looks like #37377 which should be fixed by updating `mkl-service`. Your workaround seems fine though.

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**